### ISSUE TITLE
Overlapping resize handles

### ISSUE THREAD

**Issue by AMBudnik** [handsontable/handsontable#3346]
### Description

The resize handler is overlapping the scrollbar when we perform moving of columns and rows. While it doesn't look that bad for columns, for rows and small containers it might cause some UI issues - we're not able to move the scroll with LMB so easily.

### Demo: 

http://jsfiddle.net/handsoncode/admgr5o6/ 
_(updated to `7.1.0` + `manualRowResize` enabled)_

### Reproduction
1. Press the LMB on the right border of the column to get the resize handler
2. Resize the column to the right to see that the handler is on the scrollbar
3. Release the LBM and click the same coordinates again - you'll get the handler

![col](https://cloud.githubusercontent.com/assets/10757813/13951501/b3f5cdde-f031-11e5-82f4-21b836086817.png)
![row](https://cloud.githubusercontent.com/assets/10757813/13951502/b3f92556-f031-11e5-9082-f778bfe671c3.png)

**Comment by krzysztofspilka** [comment]
@AMBudnik can't reproduce it on Windows. Could you describe the problem and how to observe it?

**Comment by AMBudnik** [comment]
@krzysztofspilka I've updated the demo: http://jsfiddle.net/3fbdhsfg/3/ (there was `manualRow` instead of `manualRowResize`). 

Column overlapping is hard to get but as you can see in second example resize handle covers nearly a half of the horizontal scroll.

To reproduce just grab the scroll in the middle (I've added more columns be be easier to perform a quick reproduction)

**Comment by AMBudnik** [comment]
Here's the second example with Chrome 48
![chrome](https://cloud.githubusercontent.com/assets/10757813/14012162/3139b044-f1a4-11e5-97a1-bce5f5a1d61a.png)


here's the Firefox 45
![ff](https://cloud.githubusercontent.com/assets/10757813/14012168/3a7eda3a-f1a4-11e5-821f-4a52cb6f1f9d.png)


surprisingly Internet Explorer seems to be not affected.

**Comment by AMBudnik** [comment]
Still present Chrome 50 / Hot v0.25.0

**Comment by AMBudnik** [comment]
Present in v0.31.0

![example11](https://cloud.githubusercontent.com/assets/10757813/23357985/6900616a-fce0-11e6-9711-d30e9ec90263.gif)

**Comment by AMBudnik** [comment]
Still replicable in v0.35.0

**Comment by AMBudnik** [comment]
The issue is replicable in v `2.0.0.` and doesn't seem to work fine in any of the previous versions.

**Comment by AMBudnik** [comment]
There's a second case where we keep the handler when using a dbclick

1. Resize any row that that handler goes under the table
2. Use double click on the handler
3. Use <kbd>RMB</kbd> on the same area

![resize](https://user-images.githubusercontent.com/10757813/68379729-b3922e00-014e-11ea-94f0-83174b2536a5.gif)


**Result**
The row is as wide as your current cursor location

**Expected result**
Nothing should happen

**Comment by pnowak** [comment]
This issue is still replicable using v8.0.0. Here's an updated demo - https://jsfiddle.net/rz5L9co6/

**Comment by AMBudnik** [comment]
This issue is no longer replicable in Chrome 106 using Handsontable v12.1.3. Updated demo http://jsfiddle.net/gr9osmw0/

In older versions, we had the resize handle stick to the same position as it was left in the container space, so once we did this

<img width="235" alt="Screenshot 2022-10-18 at 12 12 30" src="https://user-images.githubusercontent.com/10757813/196402715-336c5251-b0cf-4c29-821a-875b03082038.png">

and expanded the row height while releasing the LMB on the scroll. We could "find" the same handle even while scrolling the table

<img width="249" alt="Screenshot 2022-10-18 at 12 12 52" src="https://user-images.githubusercontent.com/10757813/196402877-a73b0c9a-e4e0-446b-898d-0ffeb96b8d92.png">

In the newest version, the handle appears only when and where it's needed.
